### ISSUE TITLE
[Feature Request]: Separate instance

### ISSUE THREAD

**Issue by Madis0** [Fabulously-Optimized/vanilla-installer#61]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Potential Enhancement

A checkbox that would install the modpack to a separate folder inside .minecraft

### How it could be implemented

A checkbox in the GUI. Folder name could be `Fabulously Optimized x.x.x` for example, where x.x.x is the Minecraft version so it can be easily updated by subsequent installs.

**Comment by osfanbuff63** [comment]
This is already supported by the file dialog to select a folder so you could select any folder anywhere - especially since from everything I've heard recently, you shouldn't nest instances inside .minecraft, you should use a different folder inside pretty much anywhere else.

We might still want to implement this, but not sure.

**Comment by Madis0** [comment]
Well, if I select a new folder outside .minecraft, e.g. .minecraft-2, it gets stuck on setting profiles.

**Comment by osfanbuff63** [comment]
> Well, if I select a new folder outside .minecraft, e.g. .minecraft-2, it gets stuck on setting profiles.

Well that's apparently a bug...you don't need to file a new ticket for it unless you want to, we have enough 😅. What's probably happening is that dir is being passed to the create profile function, which **needs** to be .minecraft since that's the only place that file exists.

**Comment by Madis0** [comment]
Indeed, the installer needs to know where .minecraft is, so I wouldn't exactly consider that a bug if I don't tell it where it is :D 

So separate profiles should be either inside or next to .minecraft, IMO letting users define the specific path/folder name is not necessary.

**Comment by tuxinal** [comment]
> What's probably happening is that dir is being passed to the create profile function, which **needs** to be .minecraft since that's the only place that file exists.

We could hardcode the .minceraft folder for the create profile function since the only way for it to need another value is to change the launcher's working directory which can only be changed through command line options. at which point we can assume the user knows enough to not be troubled by this.

**Comment by Madis0** [comment]
> We could hardcode the .minceraft folder for the create profile function since the only way for it to need another value is to change the launcher's working directory which can only be changed through command line options. at which point we can assume the user knows enough to not be troubled by this.

I don't agree with that, Minecraft Launcher can also be used as portable, in a different hard drive or various methods in Linux.
